script.js

```diff
@@ -1 +1,341 @@
-console.log('yay!');
+/* eslint-disable no-unused-vars */
+/* eslint-disable prefer-const */
+const board = [];
+let firstCard = null;
+const boardSize = 4; // has to be an even number
+const winCondition = (boardSize * boardSize) / 2;
+let deck;
+let score = 0;
+let firstSquareClick;
+let secondSquareClick;
+let abilityToClick = true;
+let totalMatched = 0;
+let gameMode = 'start';
+
+// Helper function copied
+const makeDeck = (cardAmount) => {
```

Comment:
Consider abstracting helper functions not specifically related to the Match Game into a separate JS file, e.g. `cardFunctions.js`